hazelcast/src/main/java/com/hazelcast/concurrent/lock/LockProxySupport.java

```diff
@@ -123,14 +125,16 @@ public boolean tryLock(NodeEngine nodeEngine, Data key) {
         }
     }
 
-    public boolean tryLock(NodeEngine nodeEngine, Data key, long timeout, TimeUnit timeunit) throws InterruptedException {
+    public boolean tryLock(NodeEngine nodeEngine, Data key,
+                           long timeout, @Nullable TimeUnit timeunit) throws InterruptedException {
         return tryLock(nodeEngine, key, timeout, timeunit, -1, TimeUnit.MILLISECONDS);
     }
 
-    public boolean tryLock(NodeEngine nodeEngine, Data key, long timeout, TimeUnit timeunit,
-                           long leaseTime, TimeUnit leaseTimeunit) throws InterruptedException {
-        long timeoutInMillis = getTimeInMillis(timeout, timeunit);
-        long leaseTimeInMillis = getTimeInMillis(leaseTime, leaseTimeunit);
+    public boolean tryLock(NodeEngine nodeEngine, Data key,
+                           long timeout, @Nullable TimeUnit timeunit,
+                           long leaseTime, @Nullable TimeUnit leaseTimeunit) throws InterruptedException {
```

Comment:
shouldn't these be `@Nonnull` instead of `@Nullable TimeUnit timeunit`? If you use this version of tryLock, you must provide non-null timeouts?

Reply:
I was a bit surprised myself but seems like the default is always defaulted to milliseconds by using [this method](https://github.com/hazelcast/hazelcast/blob/f638a9264d930dcc91c27b446435fb7764efefb5/hazelcast/src/main/java/com/hazelcast/concurrent/lock/LockProxySupport.java#L136-L137). Which is why I added `@Nullable`. Same for methods on `IMap`.